OpenRA.Game/Map/Map.cs

```diff
@@ -1353,13 +1353,18 @@ public IEnumerable<CPos> FindTilesInAnnulus(CPos center, int minRange, int maxRa
 				throw new ArgumentOutOfRangeException(nameof(maxRange),
 					$"The requested range ({maxRange}) cannot exceed the value of MaximumTileSearchRange ({Grid.MaximumTileSearchRange})");
 
-			for (var i = minRange; i <= maxRange; i++)
+			return FindTilesInAnnulus();
+
+			IEnumerable<CPos> FindTilesInAnnulus()
 			{
-				foreach (var offset in Grid.TilesByDistance[i])
+				for (var i = minRange; i <= maxRange; i++)
 				{
-					var t = offset + center;
-					if (allowOutsideBounds ? Tiles.Contains(t) : Contains(t))
-						yield return t;
+					foreach (var offset in Grid.TilesByDistance[i])
+					{
+						var t = offset + center;
+						if (allowOutsideBounds ? Tiles.Contains(t) : Contains(t))
+							yield return t;
+					}
```

Comment:
this is quite a weird change. Why create a new IEnumerable?

Reply:
This is [RCS1227](https://josefpihrt.github.io/docs/roslynator/analyzers/RCS1227). In the fixed form, the argument exceptions will throw immediately. The previous form doesn't trigger those exceptions until you try and enumerate the results - which might not be until much later.